### PR TITLE
ci: Improve tests pipeline runtime, group tests

### PR DESCRIPTION
**Pull request by def-**
Old tests pipeline:
Screenshot 2023-10-16 at 13 12 23
New tests pipeline:
Screenshot 2023-10-16 at 13 12 51

Old nightly pipeline:
Screenshot 2023-10-16 at 13 19 03
New nightly pipeline:
Screenshot 2023-10-16 at 13 19 40
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by nrainer-materialize — APPROVED** [review]
Awesome, I love it!

File: ci/nightly/pipeline.template.yml [hunk 1]
```diff
@@ -300,233 +306,239 @@ steps:
       - ./ci/plugins/mzcompose:
           composition: secrets-local-file
```
Comment (philip-stoev) [inline ci/nightly/pipeline.template.yml:307]: You can group the 2 secrets jobs too.

**Review by philip-stoev — APPROVED** [review]
Thank you!

**Comment by guswynn** [comment]
this is SOOO nice, thanks!

**Pull request merged**